RxSwift/Observables/Observable+Multiple.swift

```diff
@@ -62,6 +62,14 @@ extension ObservableType where E : ObservableConvertibleType {
     }
 }
 
+// switchIfEmpty
+
+extension ObservableType {
+    public func switchIfEmpty<S: ObservableConvertibleType>(resultSelector: @escaping () throws -> S) -> Observable<E> where S.E == E {
```

Comment:
Maybe we should call it.

```
public func ifEmpty(switch: Observable<E>) -> Observable<E>
```

We wouldn't lose any generality because if one wants delayed evaluation it could use:

```
xs.ifEmpty(switch: Observable.defered {
          ....
          return result
})
```

And it seems we would be more consistent with RxJava 1.x. I still need to take a look at RxJava 2.x and see what they've done there.